### PR TITLE
PLAT-421: Rewrite from workflow to composite action

### PR DESCRIPTION
**Pull request by bjarneo**
A reusable workflow did not actually work as expected. By changing the approach from workflow to a composite action, it is easier to reuse the flow while integrating with a matrix strategy.

### CODE REVIEW

**Comment by bjarneo** [comment]
Ran a test, and we have to mask some input. Looking into this.

**Comment by bjarneo** [comment]
uhm, nope, even though it is masked, we can see the output while creating the mask:
Screenshot 2022-01-24 at 11 01 42

Looking into this

File: action.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        aws-region: ${{ inputs.aws_region }}
+
+    - name: Deploy to the EKS cluster
+      shell: bash
+      env:
+        KUBE_CONFIG: ${{ inputs.KUBE_CONFIG }}
+      run: |
+        # Get the kubectl binary
+        curl -s -o kubectl https://amazon-eks.s3.us-west-2.amazonaws.com/1.20.4/2021-04-12/bin/linux/amd64/kubectl
+        chmod +x kubectl
```
Comment (lemonsaurus) [inline action.yaml:95]: So, this file is about 39MB - In the long run, we may wanna consider either caching or using a custom docker image with these tools built in for this workflow. But I don't think there's an urgent need for that.

**Review by lemonsaurus — APPROVED** [review]
This looks very nice. Well documented, very readable.

File: action.yaml (hunk 1 above)
Comment (bjarneo) [reply action.yaml:95]: Yes. Totally agree. Let us look into that. As discussed (PM), let us test azure kubectl again, and if that does not work, we will try a dockerfile.

**Pull request merged**